test/stablecoin_converter.js

```diff
@@ -223,6 +223,147 @@ contract("StablecoinConverter", async (accounts) => {
       assert.equal((orderResult2.sellAmount).toNumber(), 10, "sellAmount was stored incorrectly")
       assert.equal((orderResult2.buyAmount).toNumber(), 5, "buyAmount was stored incorrectly")
     })
+    it("places two orders and first matches them partically and then fully in a 2nd solution submission", async () => {
+      const stablecoinConverter = await StablecoinConverter.new(2**16-1)
+      const erc20_1 = await MockContract.new()
+      const erc20_2 = await MockContract.new()
+      
+      await erc20_1.givenAnyReturnBool(true)
+      await erc20_2.givenAnyReturnBool(true)
+
+      await stablecoinConverter.deposit(erc20_1.address, 10, {from: user_1})
+      await stablecoinConverter.deposit(erc20_2.address, 20, {from: user_2})
+      
+      await stablecoinConverter.addToken(erc20_1.address)
+      await stablecoinConverter.addToken(erc20_2.address)
+      const batchIndex = (await stablecoinConverter.getCurrentStateIndex.call()).toNumber()
+
+      const orderId1 = await sendTxAndGetReturnValue(stablecoinConverter.placeOrder, 1, 0, true, batchIndex + 1, 20, 10, {from: user_1})
+      const orderId2 = await sendTxAndGetReturnValue(stablecoinConverter.placeOrder, 0, 1, true, batchIndex + 1, 10, 20, {from: user_2})
+      // close auction
+      await waitForNSeconds(BATCH_TIME)
+
+      const prices = [10, 20]
+      const owner = [user_1, user_2]  //tradeData is submitted as arrays
```

Comment:
Having to write this all the time is a sign that really tradeData should be an object and we should pass an array of those.